python/tank/platform/engine.py

```diff
@@ -1256,30 +1256,34 @@ def _apply_external_styleshet(self, bundle, widget):
 
     def _define_qt_base(self):
         """
-        This will be called at initialisation time and will allow 
+        This will be called at initialisation time and will allow
         a user to control various aspects of how QT is being used
         by Tank. The method should return a dictionary with a number
-        of specific keys, outlined below. 
-        
+        of specific keys, outlined below.
+
         * qt_core - the QtCore module to use
         * qt_gui - the QtGui module to use
+        # wrapper - the Qt wrapper root module, e.g. PySide
         * dialog_base - base class for to use for Tank's dialog factory
-        
+
         :returns: dict
         """
-        # default to None
         base = {"qt_core": None, "qt_gui": None, "dialog_base": None}
         try:
-            from PySide import QtCore, QtGui
-            base["qt_core"] = QtCore
-            base["qt_gui"] = QtGui
-            base["dialog_base"] = QtGui.QDialog
+            # Local import since Qt might not be available.
+            from .qt.qt_importer import QtImporter
+            importer = QtImporter()
+            base["qt_core"] = importer.QtCore
+            base["qt_gui"] = importer.QtGui
+            base["dialog_base"] = importer.QtGui.QDialog
+            base["wrapper"] = importer.wrapper
```

Comment:
The purpose of this method is currently to provide an qt-agnostic set of entry points for app code.
The goal is to be able to do `from sgtk.platform.qt import QtCore, QtGui` and that dialog creation is handled correctly. But with varying pyside/qt installations for different engines.

It is being called from the engine constructor - https://github.com/shotgunsoftware/tk-core/blob/master/python/tank/platform/engine.py#L128:

The idea is that this is a method that is called from within the engine base class at setup, but its behaviour can be altered through subclassing if needed. Right now, multiple engines implement the method, but it is only ever called from the engine constructor. Are we extending its use now? In this context i don't totally understand the idea of the wrapper key - where is it meant to be used? And as far as I understand, i think adding a `QWidgets` key would break compatibility between qt4/qt5 setups.